src/ServicePulse.Host/vue/src/components/OrderBy.vue

```diff
@@ -1,17 +1,32 @@
+<script lang="ts">
+export function getSortFunction<T>(selector: SortOptions<T>["selector"], dir: SortDirection) {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  if (!selector) return (firstElement: T, secondElement: T) => 0;
+  return (firstElement: T, secondElement: T) => {
+    if (dir === SortDirection.Ascending) {
+      return selector(firstElement) < selector(secondElement) ? -1 : 1;
+    } else {
+      return selector(firstElement) < selector(secondElement) ? 1 : -1;
+    }
+  };
+}
+</script>
+
 <script setup lang="ts">
 import { onMounted, ref } from "vue";
 import { useCookies } from "vue3-cookies";
 import SortOptions, { SortDirection } from "@/resources/SortOptions";
-import GroupOperation from "@/resources/GroupOperation";
 
 const emit = defineEmits<{
-  sortUpdated: [option: SortOptions];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  sortUpdated: [option: SortOptions<any>];
 }>();
 
 const props = withDefaults(
   defineProps<{
     hideSort?: boolean;
-    sortOptions: SortOptions[];
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    sortOptions: SortOptions<any>[];
```

Comment:
It's possible that this component could be made generic, in which case this could be `SortOptions<T>`. I tried that once before and got bogged down but now that `SortOptions<>` has become generic it may work better.